Google.Api.Gax.Grpc/Rest/RestChannel.cs

```diff
@@ -106,22 +101,8 @@ public RestChannel(RestServiceCollection serviceCollection, string endpoint, Cha
 
             httpRequest.Headers.Add(VersionHeaderBuilder.HeaderName, RestVersion);
 
-            HttpResponseMessage httpResponseMessage;
-            try
-            {
-                await AddAuthHeadersAsync(httpRequest, restMethod, overallCancellationToken).ConfigureAwait(false);
-                httpResponseMessage = await _httpClient.SendAsync(httpRequest, httpCompletionOption, overallCancellationToken).ConfigureAwait(false);
-            }
-            catch (OperationCanceledException ex) when (deadlineToken.IsCancellationRequested)
-            {
-                throw new RpcException(new Status(StatusCode.DeadlineExceeded, $"RPC '{restMethod.FullName}' timed out", ex));
-            }
-            catch (OperationCanceledException ex)
-            {
-                throw new RpcException(new Status(StatusCode.Cancelled, $"RPC '{restMethod.FullName}' was cancelled", ex));
-            }
-
-            return httpResponseMessage;
+            await cancellationContext.AwaitAsync(token => AddAuthHeadersAsync(httpRequest, restMethod, token)).ConfigureAwait(false);
+            return await cancellationContext.AwaitAsync(token => _httpClient.SendAsync(httpRequest, httpCompletionOption, token)).ConfigureAwait(false);
```

Comment:
Maybe this is more readable? Not sure though.

```suggestion
            return await cancellationContext.AwaitAsync(token =>
            {
                await AddAuthHeadersAsync(httpRequest, restMethod, token).ConfigureAwait(false);
                return await _httpClient.SendAsync(httpRequest, httpCompletionOption, token).ConfigureAwait(false);
            }).ConfigureAwait(false);
```

Reply:
Will try a local method - I think that'll be cleaner than either.